iothub/service/src/DigitalTwin/DigitalTwinsClient.cs

```diff
@@ -0,0 +1,363 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Common.Exceptions;
+using Microsoft.Azure.Devices.Http2;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// This client contains methods to retrieve and update digital twin information, and invoke commands
+    /// on a digital twin device.
+    /// </summary>
+    /// <seealso href="https://docs.microsoft.com/en-us/azure/iot-develop/concepts-digital-twin"/>
+    public class DigitalTwinsClient
+    {
+        private string _hostName;
+        private IotHubConnectionProperties _credentialProvider;
+        private HttpClient _httpClient;
+        private HttpRequestMessageFactory _httpRequestMessageFactory;
+
+        private const string DigitalTwinRequestUriFormat = "/digitaltwins/{0}";
+        private const string DigitalTwinCommandRequestUriFormat = "/digitaltwins/{0}/commands/{1}";
+        private const string DigitalTwinComponentCommandRequestUriFormat = "/digitaltwins/{0}/components/{1}/commands/{2}";
+
+        // HttpMethod does not define PATCH in its enum in .netstandard 2.0, so this is the only way to create an
+        // HTTP patch request.
+        private readonly HttpMethod _patch = new HttpMethod("PATCH");
+
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        protected DigitalTwinsClient()
+        {
+        }
+
+        internal DigitalTwinsClient(string hostName, IotHubConnectionProperties credentialProvider, HttpClient httpClient, HttpRequestMessageFactory httpRequestMessageFactory)
+        {
+            _credentialProvider = credentialProvider;
+            _hostName = hostName;
```

Comment:
Assign params in same order as they are passed.

```suggestion
            _hostName = hostName;
            _credentialProvider = credentialProvider;
```